Backend/Remora.Discord.API/Json/Converters/Internal/MessageComponentConverter.cs

```diff
@@ -68,7 +68,11 @@ internal class MessageComponentConverter : JsonConverter<IMessageComponent>
                 => JsonSerializer.Deserialize<IActionRowComponent>(document.RootElement.GetRawText(), options),
             ComponentType.Button
                 => JsonSerializer.Deserialize<IButtonComponent>(document.RootElement.GetRawText(), options),
-            ComponentType.SelectMenu
```

Comment:
Perhaps these should be distinct types, considering they're different component types?

Reply:
Sure thing.

Reply:
Alright, I've added dedicated records and interfaces for each select component type.